src/JsonSchema/SchemaFactory.php

```diff
@@ -69,7 +69,7 @@ public function addDistinctFormat(string $format): void
      */
     public function buildSchema(string $className, string $format = 'json', string $type = Schema::TYPE_OUTPUT, ?string $operationType = null, ?string $operationName = null, ?Schema $schema = null, ?array $serializerContext = null, bool $forceCollection = false): Schema
     {
-        $schema = $schema ?? new Schema();
+        $schema = null !== $schema ? clone $schema : new Schema();
```

Comment:
This is the fix

Reply:
`$schema ? clone $schema : new Schema()` would work as well right?

Reply:
Sorry I missed this comment. Thanks for proposing a suggestion, I've just accepted it